_posts/2024-04-30-search-indexing-rollover.adoc

```diff
@@ -0,0 +1,281 @@
+---
+layout: post
+title: 'Indexing rollover with Quarkus and Hibernate Search'
+date: 2024-04-30
+tags: hibernate search howto
+synopsis: 'This is the first post in the series that dives into the implementation details of the search.quarkus.io application. Are you interested in near zero-downtime reindexing? Then this one is for you!'
+author: markobekhta
+---
+:hibernate-search-version: 7.0
+:imagesdir: /assets/images/posts/search-indexing-rollover
+:hibernate-search-docs-url: https://docs.jboss.org/hibernate/search/{hibernate-search-version}/reference/en-US/html_single/
+:quarkus-hibernate-search-docs-url: https://quarkus.io/guides/hibernate-search-orm-elasticsearch
+
+This is the first post in the series diving into the implementation details of the
+link:https://github.com/quarkusio/search.quarkus.io[application] backing the guide search of
+link:https://quarkus.io/guides/[quarkus.io].
+
+Does your application have full-text search capabilities and use Hibernate Search?
+Do you need to perform reindexing of your data while keeping your application running and producing search results?
+Look no further. In this post, we'll cover how you can approach this problem
+and solve it in practice with a few low-level APIs provided by Hibernate Search.
```

Comment:
This is a good intro, but quite restrictive as not many readers of the Quarkus blog will be in this position. They may stop reading right here.

Why not something like:

```suggestion
Does your application need full-text search capabilities? Do you need to keep your application running and producing search results without any downtime, even when reindexing all your data?
Look no further. In this post, we'll cover how you can approach this problem
and solve it in practice with a few low-level APIs, provided you use Hibernate Search, be it link:{quarkus-hibernate-search-docs-url}[on top of Hibernate ORM] or https://quarkus.io/version/main/guides/hibernate-search-standalone-elasticsearch[in standalone mode].
```